### PR TITLE
CAPZ E2E presubmit

### PR DESCRIPTION
**Pull request by jadarsie**
Adding the e2e presubmit for cluster-api-provider-azure

### CODE REVIEW

File: config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml [hunk 1]
```diff
@@ -63,6 +86,41 @@ presubmits:
...
+        hostPath:
+          path: /lib/modules
+          type: Directory
+      - name: cgroup
+        hostPath:
+          path: /sys/fs/cgroup
+          type: Directory
+    annotations:
+      testgrid-dashboards: sig-cluster-lifecycle-cluster-api-provider-azure
+      testgrid-tab-name: pr-integration
```
Comment (CecileRobertMichon) [inline config/jobs/kubernetes-sigs/cluster-api-provider-azure/cluster-api-provider-azure-presubmits.yaml]: nit: can we call this pr-e2e for consistency with the aws provider?

**Comment by jadarsie** [comment]
@awesomenix @justaugustus review please

**Comment by CecileRobertMichon** [comment]
@jadarsie please squash the commits
/hold

**Comment by jadarsie** [comment]
> @jadarsie please squash the commits
> /hold

right! done

**Pull request merged**